packages/template-retail-react-app/app/components/store-locator-modal/store-locator-content.jsx

```diff
@@ -5,22 +5,126 @@
+        })
+    }
+
+    const getGeolocationSuccess = (position) => {
+        setSearchStoresParams({
+            latitude: position.coords.latitude,
+            longitude: position.coords.longitude,
+            limit: STORE_LOCATOR_NUM_STORES_PER_LOAD
+        })
+    }
+
+    const getUserGeolocation = () => {
+        if (typeof navigator !== 'undefined' && navigator.geolocation) {
+            navigator.geolocation.getCurrentPosition(getGeolocationSuccess, getGeolocationError)
+            setUserHasSetManualGeolocation(false)
+        } else {
+            console.log('Geolocation not supported')
+        }
+    }
+
+    useEffect(() => {
+        if (!userHasSetManualGeolocation) getUserGeolocation()
+    }, [])
 
-const StoreLocatorContent = ({form, submitForm, storesInfo, searchStoresParams}) => {
+    return getUserGeolocation
+}
+
+const StoreLocatorContent = ({
+    searchStoresParams,
+    setSearchStoresParams,
+    userHasSetManualGeolocation,
+    setUserHasSetManualGeolocation
+}) => {
     const intl = useIntl()
 
+    const form = useForm({
+        mode: 'onChange',
+        reValidateMode: 'onChange',
+        defaultValues: {
+            countryCode: userHasSetManualGeolocation ? searchStoresParams.countryCode : '',
+            postalCode: userHasSetManualGeolocation ? searchStoresParams.postalCode : ''
+        }
+    })
+
+    const getUserGeolocation = useGeolocation(
+        setSearchStoresParams,
+        userHasSetManualGeolocation,
+        setUserHasSetManualGeolocation
+    )
+
+    const {data: searchStoresData, isLoading} = useSearchStores(
+        {
+            parameters: {
+                countryCode: searchStoresParams.latitude
+                    ? undefined
+                    : searchStoresParams.countryCode,
+                postalCode: searchStoresParams.latitude ? undefined : searchStoresParams.postalCode,
+                latitude: searchStoresParams.countryCode ? undefined : searchStoresParams.latitude,
+                longitude: searchStoresParams.countryCode
+                    ? undefined
+                    : searchStoresParams.longitude,
+                locale: intl.locale,
+                maxDistance: STORE_LOCATOR_DISTANCE,
+                limit: searchStoresParams.limit,
+                offset: 0,
+                distanceUnit: STORE_LOCATOR_DISTANCE_UNIT
+            }
+        },
+        {keepPreviousData: true}
+    )
+
+    const storesInfo = isLoading ? undefined : searchStoresData?.data || []
+    const numStores = searchStoresData?.total || 0
+
+    const submitForm = async (formData) => {
+        const {postalCode, countryCode} = formData
+        if (postalCode !== '' && countryCode !== '') {
+            setSearchStoresParams({
+                postalCode: postalCode,
+                countryCode: countryCode,
+                limit: STORE_LOCATOR_NUM_STORES_PER_LOAD
+            })
+            setUserHasSetManualGeolocation(true)
+        }
```

Comment:
There's no error message shown to the user if they have not filled out all of the required fields.